phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -194,13 +194,10 @@ public Tuple next() throws SQLException {
                             if (retry <= 0) {
                                 throw e1;
                             }
+                            Long cacheId = ((HashJoinCacheNotFoundException) e1).getCacheId();
                             retry--;
                             try {
-                                Long cacheId = ((HashJoinCacheNotFoundException) e1).getCacheId();
-
-                                ServerCache cache = caches == null ? null :
```

Comment:
Do we not need this null check any more?

Reply:
reverted, I think this was accidentally changed during git rebase.